Show IMEI validation status on frontend

comzeradd opened the pull request:
- Fixed #154

comzeradd committed:
Show IMEI validation status on frontend

comzeradd commented:
@akatsoulas r?

akatsoulas commented on feedthefox/base/static/js/profile.js:
```diff
@@ -4,8 +4,33 @@ jQuery(document).ready(function ($) {
+                    if (json.status) {
+                        $(status).removeClass('fa-times');
+                        $(status).addClass('fa-check');
+                        $(status).text(' valid');
```
Instead of using text and check mark ,maybe we could just change the color of the input box. Keep it blue when it is valid and change it to red while it's invalid.

akatsoulas commented:
If I enter a valid IMEI and don't submit it, then the form renders as valid any input.

akatsoulas commented:
If I submit an invalid IMEI, the the ValidationError of the Django form is displayed while the ajax request marks as valid the new input.

akatsoulas commented:
This is more of a question, but should we display the legal text while the input is not valid or just show it when the validation is complete?

akatsoulas commented on feedthefox/users/views.py:
```diff
@@ -1,11 +1,29 @@
+            response_data['status'] = 1
+        except:
+            response_data['status'] = 0
+        return HttpResponse(
```
Please use `JsonResponse` from `django.http`.

akatsoulas commented on feedthefox/users/views.py:
```diff
@@ -1,11 +1,29 @@
+        response_data = {}
+        try:
+            status = validate_imei(imei)
+            response_data['status'] = 1
```
s/1/True

akatsoulas commented on feedthefox/users/views.py:
```diff
@@ -1,11 +1,29 @@
+            status = validate_imei(imei)
+            response_data['status'] = 1
+        except:
+            response_data['status'] = 0
```
s/0/False

akatsoulas commented on feedthefox/users/views.py:
```diff
@@ -1,11 +1,29 @@
+            content_type="application/json"
+        )
+    else:
+        raise Http404
```
Instead of raising a 404 error, shouldn't we just display the form errors if exist?

comzeradd replied:
This for the case that someone access that url directly. If we don't want to forbid we should remove the if/else clause completely.

comzeradd commented:
@akatsoulas PR updated. I simplified the UI by just highlighting the input green/red, as you suggested. I also added the `autocomplete` flag to avoid refresh issues with validation.

I think it makes sense to only hide the TOS when the IMEI field is empty. Even with an invalid IMEI number a user should be able to read the terms.

akatsoulas commented:
When I am submitting non numeric characters then the input is rendered as valid. This should be happening due to the `raise Http404`.

comzeradd commented:
PR updated. This was not caused by the 404 raise on the view, but one step before on the url catch. Inserting non-numeric characters bypasses the view completely.

akatsoulas commented:
lgtm r+
